### PR TITLE
15.1 finish the backend project base code 1 revised

### PR DESCRIPTION
**Pull request by Junzhou-712**
This PR is for issue #26, 
Added language entity and built an associative relation between Entry and Language.

### CODE REVIEW

File: Backend/Underdogs-Struggling/src/main/java/com/TheTroisMousquetaires/Underdogs/Struggling/entities/Language.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+public class Language {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    @Column(name = "language_id")
+    private Long languageID;
+
+    @Column(name = "language_code", unique = true, nullable = false)
+    private String languageCode;
+    
+}
```
Comment (RunpengNie) [inline Backend/Underdogs-Struggling/src/main/java/com/TheTroisMousquetaires/Underdogs/Struggling/entities/Language.java]: Should we add one more column `language`? For example if we add English as the first language, id will be 1, code will be EN, and `language` field will be "English". What do you think of this design?

**Comment by RunpengNie** [comment]
~~Do not close the releated issue page before the PR is merged.~~
~~Please update the design document together to avoid code-desgin mismatch.~~
~~By the way I remember you also found another bug in `Entry` class. Please fix it in this PR too.~~
Resolved.

**Review by RunpengNie — CHANGES_REQUESTED** [review]

File: Backend/Underdogs-Struggling/src/main/java/com/TheTroisMousquetaires/Underdogs/Struggling/entities/Language.java (hunk 1 above)
Comment (Junzhou-712) [reply Backend/Underdogs-Struggling/src/main/java/com/TheTroisMousquetaires/Underdogs/Struggling/entities/Language.java]: Yes, you are correct. I will add a `languageName` attribute to make sure there will not be any mapping design in the front end.
And for the attribute `languageCode` will be retrieved and create a Locale class based on this code.

**Comment by Junzhou-712** [comment]
> Do not close the releated issue page before the PR is merged. Please update the design document together to avoid code-desgin mismatch. By the way I remember you also found another bug in `Entry` class. Please fix it in this PR too.

It has also been sloved.

File: Backend/Underdogs-Struggling/src/main/java/com/TheTroisMousquetaires/Underdogs/Struggling/entities/Language.java (hunk 1 above)
Comment (Junzhou-712) [reply Backend/Underdogs-Struggling/src/main/java/com/TheTroisMousquetaires/Underdogs/Struggling/entities/Language.java]: Solved

File: Backend/Underdogs-Struggling/src/main/java/com/TheTroisMousquetaires/Underdogs/Struggling/controllers/DocServiceController.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
+package com.TheTroisMousquetaires.Underdogs.Struggling.controllers;
```
Comment (RunpengNie) [inline Backend/Underdogs-Struggling/src/main/java/com/TheTroisMousquetaires/Underdogs/Struggling/controllers/DocServiceController.java]: Separate this code into another PR.

**Review by RunpengNie — CHANGES_REQUESTED** [review]
This code will not be merged until the service/controller code is separated. 1 PR SHOULD ONLY RELATED TO 1 STROY/ISSUE.

**Comment by Junzhou-712** [comment]
> This code will not be merged until the service/controller code is separated. 1 PR SHOULD ONLY RELATED TO 1 STROY/ISSUE.

Solved

**Comment by RunpengNie** [comment]
LGTM

**Review by RunpengNie — APPROVED** [review]
